gnovm/pkg/gnolang/garbage_collector.go

```diff
@@ -11,28 +34,37 @@ package gnolang
 //
 // XXX: make sure tv.T isn't bumped from allocation either.
 func (m *Machine) GarbageCollect() (left int64, ok bool) {
+	defer func() {
+		gasCPU := overflow.Mul64p(m.Alloc.visitCount*VisitCpuFactor, GasFactorCPU)
+		m.Alloc.visitCount = 0
+		if m.GasMeter != nil {
+			m.GasMeter.ConsumeGas(gasCPU, "GC")
+		}
+	}()
+
 	// We don't need the old value anymore.
 	m.Alloc.Reset()
 
 	// This is the only place where it's bumped.
-	m.gcCycle += 1
+	m.GCCycle += 1
 
 	// Construct visitor callback.
-	vis := GCVisitorFn(m.gcCycle, m.Alloc)
+	vis := GCVisitorFn(m.GCCycle, m.Alloc)
 
 	// Visit blocks
-	for block := range m.Blocks {
+	for _, block := range m.Blocks {
 		stop := vis(block)
 		if stop {
 			return -1, false
 		}
 	}
 
 	// Visit frames
-	for frame := range m.Frames {
-		// XXX implement for frames.
-		// XXX Frame is not an object,
-		// so implement a custom method and pass in vis.
+	for _, frame := range m.Frames {
+		stop := frame.Visit(vis, m.Store)
```

Comment:
the store isn't used, and should not be used for GC. no reason to hit disk.

Reply:
oh a leftover, will remove it.

Reply:
fixed.